src/main/java/org/folio/fqm/service/EntityTypeService.java

```diff
@@ -112,34 +107,24 @@ public ColumnValues getFieldValues(UUID entityTypeId, String fieldName, @Nullabl
       .findFieldDefinition(new FqlField(fieldName), entityType)
       .orElseThrow(() -> new FieldNotFoundException(entityType.getName(), fieldName));
 
-    if (field.getName().equals("tenant_id")) {
-      // Hackish way to get tenant values for cross-tenant instance queries. Can be removed after completion of MODFQMMGR-335.
-      List<String> tenants = crossTenantQueryService.getTenantsToQuery(UUID.fromString("6b08439b-4f8e-4468-8046-ea620f5cfb74"));
-      List<ValueWithLabel> tenantValues = tenants
-        .stream()
-        .map(tenant -> new ValueWithLabel().value(tenant).label(tenant))
-        .toList();
-      return new ColumnValues().content(tenantValues);
-    }
-
-    if (field.getName().equals("source_tenant_id")) {
-      // Hackish way to get tenant values for cross-tenant instance queries. Can be removed after completion of MODFQMMGR-335.
-      List<String> tenants = crossTenantQueryService.getTenantsToQuery(UUID.fromString("6b08439b-4f8e-4468-8046-ea620f5cfb74"));
-      if (tenants.size() > 1) {
-        List<ValueWithLabel> tenantValues = tenants
-          .stream()
-          .map(tenant -> new ValueWithLabel().value(tenant).label(tenant))
-          .toList();
-        return new ColumnValues().content(tenantValues);
-      } else {
-        // tenant is not central tenant, but we need to provide central tenant id as an available value
-        String centralTenantId = crossTenantQueryService.getCentralTenantId();
-        List<ValueWithLabel> tenantList = new ArrayList<>();
-        if (centralTenantId != null && !centralTenantId.equals(tenants.get(0))) {
-          tenantList.add(new ValueWithLabel().value(centralTenantId).label(centralTenantId));
+    if (field.getSource() != null) {
+      if (field.getSource().getType() == SourceColumn.TypeEnum.ENTITY_TYPE) {
+        return getFieldValuesFromEntityType(entityType, fieldName, searchText);
```

Comment:
This technically introduces a regression here, but I think it's okay.

Before, this method checked the various ways to configure the available values for a column, then just fell back to `getFieldValuesFromEntityType()`. With this change, it only does `getFieldValuesFromEntityType()` if `field.getSource()` is actually set on the column, and falls back to an exception.

I think that's okay because `getFieldValuesFromEntityType()` is very far from guaranteed to work with all columns (it depends on the entity type specifically having an "id" column), and this method only ever actually gets called when the column `source` is non-null.